azurelinuxagent/common/utils/networkutil.py

```diff
@@ -149,6 +150,13 @@ def __get_common_accept_command_params(wait, command, destination, uid):
         cmd.extend(["owner", "--uid-owner", str(uid), "-j", "ACCEPT"])
         return AddFirewallRules._add_wait(wait, cmd)
 
+    @staticmethod
+    def __get_accept_command_params_nonroot_tcp(wait, command, destination):
+        cmd = AddFirewallRules.__get_common_command_params(command, destination)
+        cmd.pop() #poped the last parameter -m since it is not needed in this rule
```

Comment:
Can you paste some doc for the `-m` parameter? Or explain a bit what it does and why its not needed here. 
Thanks!

Reply:
this should remove the -m parameter explicitly instead of relying on it always being the last command

Reply:
According to https://unix.stackexchange.com/questions/241759/what-is-the-purpose-of-m-match-in-an-iptables-rule/241790, The -m or --match option is used to enable one or more extended packet matching modules with the given name and there is not need to specify -m tcp when -p tcp is included in the rule. 
In our case
 iptables -t security -I OUTPUT 1 -d 168.63.129.16 -m tcp -p tcp  --destination-port 53 -j ACCEPT -w and
 iptables -t security -I OUTPUT 1 -d 168.63.129.16 -p tcp --destination-port 53 -j ACCEPT -w both works the same way (verified it on a VM). I went with excluding the -m option since that was the suggestion given in the bug
https://github.com/Azure/WALinuxAgent/issues/1673

Reply:
Removed -m parameter explicitly, rather than using pop